mantidimaging/core/operations/remove_stripe/test/stripe_removal_test.py:42: AssertionError

### Summary

Intermittent failure of StripeRemovalTest.test_memory_executed_wf


### Steps To Reproduce

Shows up in git hub actions tests, less that 10% of runs

### Expected Behaviour
pass

### Current Behaviour

test fails

### Context

Current master,  Seen a couple of times in past month

### Failure Logs

```
=================================== FAILURES ===================================
__________________ StripeRemovalTest.test_memory_executed_wf ___________________
[gw1] linux -- Python 3.8.8 /usr/share/miniconda/envs/mantidimaging-dev/bin/python

self = <mantidimaging.core.operations.remove_stripe.test.stripe_removal_test.StripeRemovalTest testMethod=test_memory_executed_wf>

    def test_memory_executed_wf(self):
        wf = ["level=1"]
        cached_memory = get_memory_usage_linux(kb=True)[0]
        self.do_stripe_removal(wf=wf)
>       self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
E       AssertionError: 928132.0 not less than 916150.4

mantidimaging/core/operations/remove_stripe/test/stripe_removal_test.py:42: AssertionError
```


https://github.com/mantidproject/mantidimaging/runs/2068267645

https://github.com/mantidproject/mantidimaging/runs/2288983788

--- a/docs/release_notes/next.rst
+++ b/docs/release_notes/next.rst
@@ -18,3 +18,5 @@ Fixes
 
 Developer Changes
 -----------------
+
+- #917 : Intermittent failure of StripeRemovalTest.test_memory_executed_wf

--- a/mantidimaging/core/operations/circular_mask/test/circular_mask_test.py
+++ b/mantidimaging/core/operations/circular_mask/test/circular_mask_test.py
@@ -4,11 +4,8 @@
 import unittest
 from unittest import mock
 
-import numpy as np
-
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.circular_mask import CircularMaskFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class CircularMaskTest(unittest.TestCase):
@@ -31,30 +28,6 @@ class CircularMaskTest(unittest.TestCase):
         result = CircularMaskFilter.filter_func(images, ratio)
         self.assertEqual(result.data[0, 0, 0], 0)
         self.assertEqual(result.data[0, 0, -1], 0)
-
-    def test_memory_change_acceptable(self):
-        """
-        Expected behaviour for the filter is to be done in place
-        without using more memory.
-
-        In reality the memory is increased by about 40MB (4 April 2017),
-        but this could change in the future.
-
-        The reason why a 10% window is given on the expected size is
-        to account for any library imports that may happen.
-
-        This will still capture if the data is doubled, which is the main goal.
-        """
-        images = th.generate_images()
-        ratio = 0.9
-        original = np.copy(images.data)
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        result = CircularMaskFilter.filter_func(images, ratio)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
-
-        th.assert_not_equals(result.data, original)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/clip_values/test/clip_values_test.py
+++ b/mantidimaging/core/operations/clip_values/test/clip_values_test.py
@@ -8,7 +8,6 @@ import numpy.testing as npt
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.clip_values import ClipValuesFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class ClipValuesFilterTest(unittest.TestCase):
@@ -64,31 +63,6 @@ class ClipValuesFilterTest(unittest.TestCase):
 
         npt.assert_approx_equal(result.data.min(), 0.2)
         npt.assert_approx_equal(result.data.max(), 0.8)
-
-    def test_memory_change_acceptable(self):
-        """
-        Expected behaviour for the filter is to be done in place
-        without using more memory.
-
-        In reality the memory is increased by about 40MB (4 April 2017),
-        but this could change in the future.
-
-        The reason why a 10% window is given on the expected size is
-        to account for any library imports that may happen.
-
-        This will still capture if the data is doubled, which is the main goal.
-        """
-        images = th.generate_images()
-
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        ClipValuesFilter().filter_func(images,
-                                       clip_min=0.2,
-                                       clip_max=0.8,
-                                       clip_min_new_value=0.1,
-                                       clip_max_new_value=0.9)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/crop_coords/test/crop_coords_test.py
+++ b/mantidimaging/core/operations/crop_coords/test/crop_coords_test.py
@@ -8,7 +8,6 @@ import numpy.testing as npt
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.crop_coords import CropCoordinatesFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 from mantidimaging.core.utility.sensible_roi import SensibleROI
 
 
@@ -36,32 +35,6 @@ class CropCoordsTest(unittest.TestCase):
         npt.assert_equal(result.data.shape, expected_shape)
         # check that the data has been modified
         th.assert_not_equals(result.data, sample)
-
-    def test_memory_change_acceptable(self):
-        """
-        Expected behaviour for the filter is to be done in place
-        without using more memory.
-
-        In reality the memory is increased by about 40MB (4 April 2017),
-        but this could change in the future.
-
-        The reason why a 10% window is given on the expected size is
-        to account for any library imports that may happen.
-
-        This will still capture if the data is doubled, which is the main goal.
-        """
-        images = th.generate_images()
-        roi = SensibleROI.from_list([1, 1, 5, 5])
-
-        cached_memory = get_memory_usage_linux(mb=True)[0]
-
-        result = CropCoordinatesFilter.filter_func(images, roi)
-
-        self.assertLess(get_memory_usage_linux(mb=True)[0], cached_memory * 1.1)
-
-        expected_shape = (10, 4, 4)
-
-        npt.assert_equal(result.data.shape, expected_shape)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/gaussian/test/gaussian_test.py
+++ b/mantidimaging/core/operations/gaussian/test/gaussian_test.py
@@ -8,7 +8,6 @@ import numpy as np
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.gaussian import GaussianFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class GaussianTest(unittest.TestCase):
@@ -48,30 +47,6 @@ class GaussianTest(unittest.TestCase):
         result = GaussianFilter.filter_func(images, size, mode, order)
 
         th.assert_not_equals(result.data, original)
-
-    def test_memory_change_acceptable(self):
-        """
-        Expected behaviour for the filter is to be done in place
-        without using more memory.
-
-        In reality the memory is increased by about 40MB (4 April 2017),
-        but this could change in the future.
-
-        The reason why a 10% window is given on the expected size is
-        to account for any library imports that may happen.
-
-        This will still capture if the data is doubled, which is the main goal.
-        """
-        images = th.generate_images()
-        size = 3
-        mode = 'reflect'
-        order = 1
-
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        GaussianFilter.filter_func(images, size, mode, order)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/median_filter/test/median_filter_test.py
+++ b/mantidimaging/core/operations/median_filter/test/median_filter_test.py
@@ -10,7 +10,6 @@ import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.data.images import Images
 from mantidimaging.core.gpu import utility as gpu
 from mantidimaging.core.operations.median_filter import MedianFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 GPU_UTIL_LOC = "mantidimaging.core.gpu.utility.gpu_available"
 
@@ -71,29 +70,6 @@ class MedianTest(unittest.TestCase):
         original = np.copy(images.data[0])
         result = MedianFilter.filter_func(images, size, mode)
         th.assert_not_equals(result.data, original)
-
-    def test_memory_change_acceptable(self):
-        """
-        Expected behaviour for the filter is to be done in place
-        without using more memory.
-
-        In reality the memory is increased by about 40MB (4 April 2017),
-        but this could change in the future.
-
-        The reason why a 10% window is given on the expected size is
-        to account for any library imports that may happen.
-
-        This will still capture if the data is doubled, which is the main goal.
-        """
-        images = th.generate_images()
-        size = 3
-        mode = 'reflect'
-
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        MedianFilter.filter_func(images, size, mode)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/rebin/test/rebin_test.py
+++ b/mantidimaging/core/operations/rebin/test/rebin_test.py
@@ -9,7 +9,6 @@ import numpy.testing as npt
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.rebin import RebinFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class RebinTest(unittest.TestCase):
@@ -116,29 +115,6 @@ class RebinTest(unittest.TestCase):
         # something very huge that shouldn't fit on ANY computer
         rebin_param = (100000, 100000)
         self.assertRaises(RuntimeError, RebinFilter.filter_func, images, rebin_param=rebin_param, mode=mode)
-
-    def test_memory_change_acceptable(self):
-        """
-        This filter will increase the memory usage as it has to allocate memory
-        for the new resized shape
-        """
-        images = th.generate_images()
-
-        mode = 'reflect'
-        # This about doubles the memory. Value found from running the test
-        val = 100.
-
-        expected_x = int(images.data.shape[1] * val)
-        expected_y = int(images.data.shape[2] * val)
-
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        result = RebinFilter.filter_func(images, val, mode)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 2)
-
-        npt.assert_equal(result.data.shape[1], expected_x)
-        npt.assert_equal(result.data.shape[2], expected_y)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/remove_stripe/test/stripe_removal_test.py
+++ b/mantidimaging/core/operations/remove_stripe/test/stripe_removal_test.py
@@ -6,7 +6,6 @@ from unittest import mock
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.remove_stripe import StripeRemovalFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class StripeRemovalTest(unittest.TestCase):
@@ -34,18 +33,6 @@ class StripeRemovalTest(unittest.TestCase):
     def test_executed_sf(self):
         sf = ['size=5']
         self.do_stripe_removal(sf=sf)
-
-    def test_memory_executed_wf(self):
-        wf = ["level=1"]
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-        self.do_stripe_removal(wf=wf)
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
-
-    def test_memory_executed_sf(self):
-        sf = ['size=5']
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-        self.do_stripe_removal(sf=sf)
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/ring_removal/test/ring_removal_test.py
+++ b/mantidimaging/core/operations/ring_removal/test/ring_removal_test.py
@@ -9,7 +9,6 @@ import numpy.testing as npt
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.ring_removal import RingRemovalFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class RingRemovalTest(unittest.TestCase):
@@ -30,17 +29,6 @@ class RingRemovalTest(unittest.TestCase):
         original = np.copy(images.data[0])
         result = RingRemovalFilter.filter_func(images, run_ring_removal, cores=1)
         npt.assert_equal(result.data[0], original)
-
-    def test_memory_change_acceptable(self):
-        images = th.generate_images()
-        # invalid threshold
-        run_ring_removal = False
-
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        RingRemovalFilter.filter_func(images, run_ring_removal, cores=1)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

--- a/mantidimaging/core/operations/rotate_stack/test/rotate_stack_test.py
+++ b/mantidimaging/core/operations/rotate_stack/test/rotate_stack_test.py
@@ -8,7 +8,6 @@ import numpy.testing as npt
 
 import mantidimaging.test_helpers.unit_test_helper as th
 from mantidimaging.core.operations.rotate_stack import RotateFilter
-from mantidimaging.core.utility.memory_usage import get_memory_usage_linux
 
 
 class RotateStackTest(unittest.TestCase):
@@ -36,30 +35,6 @@ class RotateStackTest(unittest.TestCase):
         result = RotateFilter.filter_func(images, rotation, cores=cores)
 
         npt.assert_equal(result.data[:, :, -1], 42)
-
-    def test_memory_change_acceptable(self):
-        """
-        Expected behaviour for the filter is to be done in place
-        without using more memory.
-
-        In reality the memory is increased by about 40MB (4 April 2017),
-        but this could change in the future.
-
-        The reason why a 10% window is given on the expected size is
-        to account for any library imports that may happen.
-
-        This will still capture if the data is doubled, which is the main goal.
-        """
-        # only works on square images
-        images = th.generate_images((10, 10, 10))
-        rotation = 1  # once clockwise
-        images.data[:, 0, 0] = 42  # set all images at 0,0 to 42
-
-        cached_memory = get_memory_usage_linux(kb=True)[0]
-
-        RotateFilter.filter_func(images, rotation)
-
-        self.assertLess(get_memory_usage_linux(kb=True)[0], cached_memory * 1.1)
 
     def test_execute_wrapper_return_is_runnable(self):
         """

